src/components/forms/EditForm.tsx

```diff
@@ -37,73 +38,87 @@ const EditForm = () => {
   } = useForm<FormValues>({ mode: "onChange", values: data });
 
   const updatePlant: SubmitHandler<FormValues> = (newData) => {
-    pb.collection(PLANTS_COLLECTION)
-      .update(plantId, {
-        ...newData,
-      })
-      .then(() => {
-        toast.success("Zmiany zostały zapisane!");
-        reset();
-        navigateTo(`/${plantId}`);
-      });
+    return pb.collection(PLANTS_COLLECTION).update(plantId, {
+      ...newData,
+    });
   };
 
-  const { mutate } = useMutation({
+  const updatePlantMutation = useMutation({
     mutationFn: (newPlantData: FormValues) => updatePlant(newPlantData),
+    onSuccess: () => {
+      toast.success("Zmiany zostały zapisane!");
+      reset();
+      navigateTo(`/${plantId}`);
+
+      return async () => {
+        await queryClient.invalidateQueries({ queryKey: [plantQueryKey] });
```

Comment:
ok, all side effects are now defined in one place! Just to spice things up - there's one more way to manipulate the cache - instead of invalidating and causing GET to be called once again, in some circumstances you can directly update the cache with new data. No GET is necessary then - react query uses data you provided.  Just FYI :)